### PR TITLE
fix/1305 logging and monitoring config

### PR DESCRIPTION
**Pull request by rwong2888**
fix [1305](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/1305)

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@rwong2888
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by rwong2888** [comment]
@Jberlinsky , @bharathkkb mind taking a look at this?

File: autogen/main/cluster.tf.tmpl [hunk 1]
```diff
@@ -119,6 +121,25 @@ resource "google_container_cluster" "primary" {
   {% endif %}
 {% endif %}
   {% if autopilot_cluster != true %}
+  {% if beta_cluster != true %}
+  # only one of logging/monitoring_service or logging/monitoring_config can be specified
+  logging_service    = local.logmon_config_is_set ? null : var.logging_service
```
Comment (bharathkkb) [inline autogen/main/cluster.tf.tmpl:126]: We already have this in 
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/4bf0011202667b6efe8a0b5aea2910f1c3250968/autogen/main/cluster.tf.tmpl#L81-L105
We should just change the conditional.

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for the PR @rwong2888

File: autogen/main/cluster.tf.tmpl (hunk 1 above)
Comment (rwong2888) [reply autogen/main/cluster.tf.tmpl:126]: I wasn't sure about "cluster_telemetry_type_is_set", "managed_prometheus", and "monitoring_enable_managed_prometheus" as I was primarily looking for "logging_config", but I can revise the PR to include those as well if you think they are ready to promote out of beta @bharathkkb

Comment (rwong2888) [reply autogen/main/cluster.tf.tmpl:126]: @bharathkkb, `cluster_telemetry` and `managed_prometheus` are still in beta as per below, so it does not look possible to simply modify the conditional. Please advise.
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster

Comment (bharathkkb) [reply autogen/main/cluster.tf.tmpl:126]: @rwong2888 Can we try doing it inline and guard the dynamic block for managed_prometheus with a conditional. Similarly for the `local.cluster_telemetry_type_is_set || local.logmon_config_is_set` conditional.

**Pull request closed without merging**